store/src/java/com/zimbra/cs/datasource/imap/RemoteFolder.java

```diff
@@ -71,10 +77,41 @@ public void delete() throws IOException {
 
     public RemoteFolder renameTo(String newName) throws IOException {
         info("renaming folder to '%s'", newName);
-        connection.rename(path, newName);
+
+        /*
+         * ZBUG-2192 execute only when last folder name is different don't change any
+         * parent folder names using "path" as base name, we are forming new folder name
+         * in such a way only last folder get's renamed at a time
+         */
+        if (!isSameFolderName(newName)) {
+            connection.rename(path, createBaseNewPath(newName));
+        }
         return new RemoteFolder(connection, newName);
     }
 
+    /*
+     * ZBUG-2192: The helper method to make comparison between old and new folder
+     * name and returns boolean value
+     */
+    private boolean isSameFolderName(String newPathName) {
+        String oldPathName = path.substring(path.lastIndexOf(PATH_SEPARATOR) + 1);
+        newPathName = newPathName.substring(newPathName.lastIndexOf(PATH_SEPARATOR) + 1);
+        return oldPathName.equals(newPathName);
+    }
+    
+    //ZBUG-2192 :Helper method to form and return newpath
```

Comment:
* add a white space `// ZBUG...`
* fix position of ":"  `// ZBUG-2192: Helper ...`